python/ray/tests/test_dynres.py

```diff
@@ -569,3 +569,25 @@ def delete_res(resource_name, res_client_id):
             all_resources_created.append(str(i) in resources)
         success = all(all_resources_created)
     assert success
+
+
+def test_release_cpus_when_actor_creation_task_blocking(shutdown_only):
+    ray.init(num_cpus=2)
+
+    @ray.remote(num_cpus=1)
+    def get_100():
+        import time
+        time.sleep(2)
+        return 100
+
+    @ray.remote(num_cpus=1)
+    class A(object):
+        def __init__(self):
+            self.num = ray.get(get_100.remote())
+
+        def get_num(self):
+            return self.num
+
+    a = A.remote()
+    assert 100 == ray.get(a.get_num.remote())
+    assert 1 == ray.available_resources()["CPU"]
```

Comment:
Perhaps I'm missing something, but why should available CPUs be 1 instead of 2 when this assertion is checked? The `ray.get(a.get_num.remote())` in the previous line ensures that both `__init__` and `get_num` have completed, thus all two CPUs should be available.

Do you intend to check this assert when the `__init__` is blocked on `get_100` at L586? If so, we should perhaps swap the order and check `100 == ray.get(a.get_num.remote())` after this assert.

Reply:
> Perhaps I'm missing something, but why should available CPUs be 1 instead of 2 when this assertion is checked? The ray.get(a.get_num.remote()) in the previous line ensures that both __init__ and get_num have completed, thus all two CPUs should be available.

The resources of an actor shouldn't be released when the actor creation task completed. That's why the final state is that the available CPU should be 1.

> Do you intend to check this assert when the __init__ is blocked on get_100 at L586? If so, we should perhaps swap the order and check 100 == ray.get(a.get_num.remote()) after this assert.

I don't want to do any check when the `__init__` is blocking. Just check the final available resource of the cluster.

Reply:
Ah sorry, I misunderstood some bits. Thanks for clarifying!